File: fm-dddrive/src/main/java/io/dddrive/core/property/model/EnumProperty.java
package io.dddrive.core.property.model;

import io.dddrive.core.enums.model.Enumerated;
import io.dddrive.core.enums.model.Enumeration;

public interface EnumProperty<E extends Enumerated> extends BaseProperty<E> {

	Enumeration<E> getEnumeration();

}

File: fm-dddrive/src/main/java/io/dddrive/core/property/model/EnumProperty.kt
package io.dddrive.core.property.model

import io.dddrive.core.enums.model.Enumerated
import io.dddrive.core.enums.model.Enumeration

interface EnumProperty<E : Enumerated> : ReferenceProperty<E, String> {

	val enumeration: Enumeration<E>

}
